src/core/src/op/util/fft_base.cpp

```diff
@@ -38,164 +39,22 @@ void ov::op::util::FFTBase::validate() {
                           axes_et == element::i64 || axes_et == element::i32,
                           "FFT op axes element type must be i32 or i64");
 
-    const auto& input_shape = PartialShape(get_input_partial_shape(0));
-    if (input_shape.rank().is_static()) {
-        const auto input_rank = input_shape.rank().get_length();
-        NODE_VALIDATION_CHECK(this,
-                              input_rank >= 2,
-                              "The input rank must be greater or equal to 2. Got input rank: ",
-                              input_rank);
-
-        auto last_dim_with_two = input_shape[input_rank - 1] & Dimension(2);
-        NODE_VALIDATION_CHECK(this,
-                              !last_dim_with_two.get_interval().empty(),
-                              "The last dimension of input data must be 2. Got: ",
-                              input_shape[input_rank - 1]);
-    }
-
-    const auto& axes_shape = PartialShape(get_input_partial_shape(1));
-    if (axes_shape.rank().is_static()) {
-        NODE_VALIDATION_CHECK(this,
-                              axes_shape.rank().get_length() == 1,
-                              "FFT op axes input must be 1D tensor. Got axes input rank: ",
-                              axes_shape.rank().get_length());
-    }
-
-    if (input_shape.rank().is_static() && axes_shape.is_static()) {
-        const auto input_rank = input_shape.rank().get_length();
-        NODE_VALIDATION_CHECK(this,
-                              input_rank >= static_cast<int64_t>(axes_shape.to_shape()[0] + 1),
-                              "The input rank must be greater than number of FFT op axes. Got "
-                              "input rank: ",
-                              input_rank,
-                              ", number of axes: ",
-                              axes_shape.to_shape()[0]);
-    }
-
-    if (input_shape.rank().is_static() && ov::is_type<ngraph::op::Constant>(input_value(1).get_node())) {
-        const auto input_rank = input_shape.rank().get_length();
-        const auto& const_axes = get_constant_from_source(input_value(1));
-        auto axes = const_axes->cast_vector<int64_t>();
-
-        // FFT operation supports for negative axes to transform. More precisely, according to
-        // the FFT operation specification, axes should be integers from -(r - 1) to (r - 2)
-        // inclusively, where r = rank(data). A negative axis 'a' is interpreted as an axis
-        // 'r - 1 + a'. The reason is the following: real input tensor of the shape
-        // [n_0, ..., n_{r - 1}, 2] is interpreted as a complex tensor with the shape
-        // [n_0, ..., n_{r - 1}].
```

Comment:
Does it make sense to move this comment to the appropriate place in the shape_infer function?

Reply:
Added comments.